### PR TITLE
media-libs/x265: Fix build on x32 by disabling assem…

### PR DESCRIPTION
**Pull request by bjacquin**
…bly code, patch to disable automagic march setup, bug #510890

Package-Manager: portage-2.2.20.1

### CODE REVIEW

**Comment by mgorny** [comment]
Commit message is too long. Put very short explanation on summary line, then explain the whole thing in detail in body, i.e.:

```
media-libs/x265: Fix x32 build

Fix build on x32 ...
```

**Comment by mgorny** [comment]
@gentoo/video

**Comment by bjacquin** [comment]
Pull requested updated as requested, upstream report made in https://github.com/videolan/x265/pull/1

**Comment by mgorny** [comment]
Errr, sorry for being picky, but I think pasting the build log in commit message is not a good idea ;-). Word the minimal description someone may need to understand the change. In particular, the description you gave on upstream commit is better than the one here ;-).

**Comment by bjacquin** [comment]
Upstream videolan/x265#1 is not the official repo, report was made on https://bitbucket.org/multicoreware/x265/pull-requests/21/build-disable-march-selection-from/diff

**Comment by bjacquin** [comment]
Can you tell me if this is better now ?

**Comment by mgorny** [comment]
Yes, looks good. Just to be clear, what happens when you enable assembly on x32? Does it fail? I suggest you file a bug upstream about that too, they may want to update their asm, or implicitly disable it when unsupported ABI is used.

**Comment by bjacquin** [comment]
Build fails, as reported upstream: https://bitbucket.org/multicoreware/x265/issues/198/assembly-code-fails-on-x32

**Comment by bjacquin** [comment]
PR updaded for x265 1.8

**Comment by aballier** [comment]
lgtm, thanks!

just one nitpick: it'd be good to have the bug number in a comment close to where you disable asm, as in my original suggestion: Unless asm optimizations are never intended to work on x32, I imagine a few years from now, people will wonder why this is disabled without warning :)

File: media-libs/x265/x265-1.8.ebuild [hunk 1]
```diff
@@ -54,6 +58,8 @@ multilib_src_configure() {
 	if [ "${ABI}" = x86 ] ; then
 		use 10bit && ewarn "Disabling 10bit support on x86 as it does not build (or requires to disable assembly optimizations)"
 		mycmakeargs+=( -DHIGH_BIT_DEPTH=OFF )
+	elif [ "${ABI}" = x32 ] ; then
```
Comment (mgorny) [inline media-libs/x265/x265-1.8.ebuild]: If I may add my own nitpick, please use bash `[[ ... ]]`. Then quoting becomes unnecessary:

    elif [[ ${ABI} = x32 ]] ; then

If that wouldn't be a problem, please also change the other conditional.

Comment (bjacquin) [inline media-libs/x265/x265-1.8.ebuild]: PR updated as per your comment

**Comment by bjacquin** [comment]
PR updated as per your comment

**Comment by aballier** [comment]
if you wish to update per above nitpicks, feel free, otherwise it's good to go


and I now see why you were using MULTILIB_ABI_FLAG: it is used in src_test...

**Comment by mgorny** [comment]
@bjacquin, your PGP key still seems to be missing on sks keyservers ;-). Please upload it. (`gpg --keyserver keys.gnupg.net  --send-key YOUR-KEY-ID`)

**Comment by bjacquin** [comment]
Hi Michał,

I'm using a EdDSA GnuPG key and therefore keys.gnupg.net does not 
support it, but the key is available here: 
https://sks-keyservers.net/pks/lookup?op=get&search=0xA3B5C016618D9AAA

Cheers,
Bertrand

On 14/10/2015 22:10, Michał Górny wrote:
> @bjacquin [1], your PGP key still seems to be missing on sks
> keyservers ;-). Please upload it. (gpg --keyserver keys.gnupg.net
> --send-key YOUR-KEY-ID)
> 
> --
> Reply to this email directly or view it on GitHub [2].
> 
> 	*
> 
> Links:
> ------
> [1] https://github.com/bjacquin
> [2] https://github.com/gentoo/gentoo/pull/173#issuecomment-148201415

-- 
Bertrand

**Pull request merged**